Stratis.Bitcoin/BlockStore/BlockStore.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NBitcoin;
+using NBitcoin.Protocol;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.BlockStore
+{
+    public class BlockStore
+    {
+	    private readonly ConcurrentChain concurrentChain;
+	    private readonly BlockRepository blockRepository;
+	    private readonly TaskFactory taskFactory;
+
+	    public BlockStore(ConcurrentChain concurrentChain, BlockRepository blockRepository)
+	    {
+		    this.concurrentChain = concurrentChain;
+		    this.blockRepository = blockRepository;
+			this.taskFactory = new TaskFactory(new CustomThreadPoolTaskScheduler(2, 500, "BlockStore"));
+	    }
+
+	    public Task ProcessGetData(Node node, GetDataPayload getDataPayload)
+	    {
+		    return this.taskFactory.StartNew(async () =>
```

Comment:
Another thing to consider is a flag on the CustomThreadPoolTaskScheduler that releases threads when there is no work.
Currently I specified 2 threads and they are blocked til there is work.